src/test/java/com/wcc/platform/integrationtests/SurrealDbIntegrationTest.java

```diff
@@ -1,52 +1,92 @@
-
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
-import org.springframework.test.context.DynamicPropertyRegistry;
-import org.springframework.test.context.DynamicPropertySource;
+import static com.wcc.platform.factories.SetupFactories.createLinkTest;
+import static com.wcc.platform.factories.SetupFactories.createNetworksTest;
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.surrealdb.connection.SurrealConnection;
+import com.surrealdb.connection.SurrealWebSocketConnection;
+import com.surrealdb.driver.SyncSurrealDriver;
+import com.wcc.platform.domain.cms.attributes.LabelLink;
+import com.wcc.platform.domain.cms.attributes.Network;
+import com.wcc.platform.domain.cms.pages.FooterPage;
+import com.wcc.platform.repository.surrealdb.SurrealDbPageRepository;
+import java.util.Collection;
+import java.util.List;
+import org.junit.jupiter.api.*;
 import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
-import org.testcontainers.utility.DockerImageName;
 
 @Testcontainers
 class SurrealDbIntegrationTest {
-
-  protected static final GenericContainer<?> SURREAL_DB_CONTAINER =
-      new GenericContainer<>(DockerImageName.parse("surrealdb/surrealdb:latest"))
+  @Container
+  static final GenericContainer<?> surrealDbContainer =
+      new GenericContainer<>("surrealdb/surrealdb:latest")
           .withExposedPorts(8000)
-          .withCommand("start --user root --pass root");
-
-  @DynamicPropertySource
-  static void registerSurrealDbProperties(final DynamicPropertyRegistry registry) {
-    SURREAL_DB_CONTAINER.start();
-    String host = SURREAL_DB_CONTAINER.getHost();
-    Integer port = SURREAL_DB_CONTAINER.getMappedPort(8000);
-    registry.add("surrealdb.host", () -> host);
-    registry.add("surrealdb.port", port::toString);
-    registry.add("surrealdb.username", () -> "root");
-    registry.add("surrealdb.password", () -> "root");
-    registry.add("surrealdb.namespace", () -> "test_namespace");
-    registry.add("surrealdb.database", () -> "test_db");
-  }
+          .withCommand("start", "--log", "debug", "--user", "root", "--pass", "password");
+
+  static final String TABLE = "page";
+  private static SurrealConnection connection;
+  private static SyncSurrealDriver driver;
+  private SurrealDbPageRepository repository;
 
   @BeforeAll
-  static void setUp() {
-    SURREAL_DB_CONTAINER.start();
+  static void setUpContainer() {
+    surrealDbContainer.start();
+
+    // Initialize SyncSurrealDriver with container connection
+    String host = surrealDbContainer.getHost();
+    Integer port = surrealDbContainer.getFirstMappedPort();
+    // String url = "ws://" + host + ":" + port + "/rpc";
+
+    connection = new SurrealWebSocketConnection(host, port, false);
```

Comment:
I think here you should be using the Bean that initialize the connection, what do you think?

Reply:
Using the Bean may make connecting take more time, and cause initializationError in Build and test as seen in the picture you showed me. To solve the errors, I just changed test timeout in build.gradle to 5 minutes. If we use the Bean, errors would still occur.